src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,158 @@
+        payment.token = dummyToken;
+        payment.amounts = amounts;
+
+        uint256 balanceBefore = dummyToken.balanceOf(address(paymentCoordinator));
+        uint256 cumumlativeEarningsBefore = paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken);
+        cheats.expectEmit(address(paymentCoordinator));
+        emit PaymentReceived(address(this), payment);
+        paymentCoordinator.makePayment(payment);
+        require(paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken) - cumumlativeEarningsBefore == sum * eigenLayerShareBIPs / MAX_BIPS, "cumulativeEigenLayerTokeEarnings should be set");
+        require(dummyToken.balanceOf(address(paymentCoordinator)) - balanceBefore == sum, "incorrect token balance");
+    }
+
+    function testPostMerkleRootFromUnauthorizedAddress(address unauthorizedAddress) external {
+        cheats.assume(unauthorizedAddress != paymentCoordinator.rootPublisher());
+        cheats.startPrank(unauthorizedAddress);
+        cheats.expectRevert(bytes("PaymentCoordinator: Only rootPublisher"));
+        paymentCoordinator.postMerkleRoot(bytes32(0), 0, 0);
+        cheats.stopPrank();
+    }
+
+    function testPostMerkleRoot(bytes32 root, uint256 height, uint256 calculatedUpToBlockNumber) public {
+        PaymentCoordinator.MerkleRootPost memory merkleRootPost;
+        merkleRootPost.root = root;
+        merkleRootPost.height = height;
+        merkleRootPost.calculatedUpToBlockNumber = calculatedUpToBlockNumber;
+
+        uint256 numMerkleRootsBefore = paymentCoordinator.merkleRootsLength();
+
+        cheats.startPrank(paymentCoordinator.rootPublisher());
+        paymentCoordinator.postMerkleRoot(root, height, calculatedUpToBlockNumber);
+        cheats.stopPrank();
+
+        (bytes32 rootPost, uint256 heightPost, uint256 confirmedAtBlockNumber, uint256 calculatedUpToBlockNumberPost) = paymentCoordinator.merkleRoots(numMerkleRootsBefore);
+        require(rootPost == root, "root should be set");
+        require(heightPost == height, "height should be set");
+        require(confirmedAtBlockNumber == block.number + rootPostDelay, "confirmedAtBlockNumber should be set");
+        require(calculatedUpToBlockNumberPost == calculatedUpToBlockNumber, "calculatedUpToBlockNumber should be set");
+    }
+
+    function testWithdrawEigenLayerShare(uint256[] memory amounts, address recipient) external {
+        cheats.assume(recipient != address(0));
+        testMakePayment(amounts);
+        uint256 amountToClaim = paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken);
+        cheats.startPrank(paymentCoordinator.rootPublisher());
+        paymentCoordinator.withdrawEigenlayerShare(dummyToken, recipient);
+        cheats.stopPrank();
+        require(paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken) == 0, "cumulativeEigenLayerTokeEarnings not updated correctly");
+        require(dummyToken.balanceOf(recipient) == amountToClaim, "incorrect token balance");
```

Comment:
presumably want to check how the recipient's balance changes, rather than pure equivalence. also probably want to filter out the recipient being the paymentCoordinator itself or else this check will still break.

Reply:
yup good point